lib/Backend/IR.cpp

```diff
@@ -2712,16 +2712,27 @@ Instr::GetNextByteCodeInstr() const
         }
         return nextInstr->GetNextRealInstrOrLabel();
     };
-    while (nextInstr->GetByteCodeOffset() == Js::Constants::NoByteCodeOffset ||
-        nextInstr->GetByteCodeOffset() == currentOffset)
+    if (currentOffset == Js::Constants::NoByteCodeOffset)
     {
-        nextInstr = getNext(nextInstr);
+        while (nextInstr->GetByteCodeOffset() == Js::Constants::NoByteCodeOffset)
+        {
+            nextInstr = getNext(nextInstr);
+        }
+        AssertMsg(nextInstr->GetByteCodeOffset() == 0, "Only instrs before the first one are allowed to not have a bytecode offset");
```

Comment:
Is there any situation in which we call this where we may have removed all instructions with bytecodeoffset 0, or may call this with some instruction with NoByteCodeOffset that is between two other instructions with non-zero bytecodeoffset?

Reply:
Maybe, but that would be a bug and we want to catch those.
The bug being that instr should have a bytecode offset. We haven't been very consistent with making sure all instrs have bytecode offset in the past. 
Normally, the only instrs that can have a bytecode offset is in the function prologue